memgpt/agent.py

```diff
@@ -49,33 +50,86 @@
 from .errors import LLMError
 
 
-def construct_system_with_memory(
-    system: str,
-    memory: BaseMemory,
-    memory_edit_timestamp: str,
+def compile_memory_metadata_block(
```

Comment:
@sarahwooders a question perhaps best saved for a later PR - do we want to move this metadata construction into the memory modules?

basically, when we compile the system prompt, there are two strings that get dynamically created related to memory, the core memory string (`str(memory)`) and metadata about archival/recall. This function (`compile_memory_metadata_block`) is generating the metadata, but I'm not sure where we want to put this.

TBH probably fine to leave in here for now, and refactor again in the PR where we introduce the ability to add user variables to the compilation process.